apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/modules/components/Mintable.tsx

```diff
@@ -0,0 +1,477 @@
+import { Textarea } from "@/components/ui/textarea";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useMutation } from "@tanstack/react-query";
+import { CircleAlertIcon } from "lucide-react";
+import { useCallback } from "react";
+import { useForm } from "react-hook-form";
+import { toast } from "sonner";
+import { type PreparedTransaction, sendAndConfirmTransaction } from "thirdweb";
+import { MintableERC721, MintableERC1155 } from "thirdweb/modules";
+import { useReadContract } from "thirdweb/react";
+import type { NFTMetadataInputLimited } from "types/modified-types";
+import { parseAttributes } from "utils/parseAttributes";
+import { z } from "zod";
+import { addressSchema } from "../zod-schemas";
+import { ModuleCardUI, type ModuleCardUIProps } from "./module-card";
+import type { ModuleInstanceProps } from "./module-instance";
+import { AdvancedNFTMetadataFormGroup } from "./nft/AdvancedNFTMetadataFormGroup";
+import { NFTMediaFormGroup } from "./nft/NFTMediaFormGroup";
+import { PropertiesFormControl } from "./nft/PropertiesFormControl";
+
+export type UpdateFormValues = {
+  primarySaleRecipient: string;
+};
+
+// TODO - add form validation with zod schema for mint form
+
+export type MintFormValues = NFTMetadataInputLimited & {
+  useNextTokenId: boolean;
+  recipient: string;
+  amount: number;
+  supply: number;
+  customImage: string;
+  customAnimationUrl: string;
+  attributes: { trait_type: string; value: string }[];
+  tokenId?: string;
+};
+
+function MintableModule(props: ModuleInstanceProps) {
+  const { contract, ownerAccount } = props;
+
+  const primarySaleRecipientQuery = useReadContract(
+    MintableERC721.getSaleConfig,
+    {
+      contract: contract,
+    },
+  );
+
+  const isErc721 = props.contractInfo.name === "MintableERC721";
+  const isBatchMetadataInstalled = !!props.allModuleContractInfo.find(
+    (module) => module.name.includes("BatchMetadata"),
+  );
+  const isSequentialTokenIdInstalled = !!props.allModuleContractInfo.find(
+    (module) => module.name.includes("SequentialTokenId"),
+  );
+
+  const mint = useCallback(
+    async (values: MintFormValues) => {
+      const nft = parseAttributes(values);
+      if (!ownerAccount) {
+        throw new Error("Not an owner account");
+      }
+
+      let mintTx: PreparedTransaction;
+      if (isErc721) {
+        mintTx = MintableERC721.mintWithRole({
+          contract,
+          to: values.recipient,
+          nfts: [nft],
+        });
+      } else if (values.useNextTokenId || values.tokenId) {
+        mintTx = MintableERC1155.mintWithRole({
+          contract,
+          to: values.recipient,
+          amount: BigInt(values.amount),
+          // biome-ignore lint/style/noNonNullAssertion: if useNextTokenId is false, then tokenId should be defined
+          tokenId: values.useNextTokenId ? undefined : BigInt(values.tokenId!),
+          nft,
+        });
+      } else {
+        throw new Error("Invalid token ID");
```

Comment:
This should be caught before the form is submitted - need to add form validation here

Reply:
Since it's not using zod as the resolver for `react-hook-forms` will have to compromise with this code
```
    const values = form.getValues();
    if (!props.isErc721 && !values.useNextTokenId && !values.tokenId) {
      form.setError("tokenId", { message: "Token ID is required" });
      return;
    }
 ```
 
 Once the zod resolver is switched over, can implement this style of validation with tokenId with respect to `useNextTokenId` 
 https://github.com/colinhacks/zod/discussions/988

Reply:
Its fine if we don't have all validations in the schema and setting errors manually like this, but whatever we can validate with zod, we should. so we can use both the zod resolver and custom validation logic